src/bpftrace.cpp

```diff
@@ -650,38 +674,22 @@ std::vector<std::unique_ptr<AttachedProbe>> BPFtrace::attach_probe(
 {
   std::vector<std::unique_ptr<AttachedProbe>> ret;
 
-  try {
-    const auto &program = bytecode.getProgramForProbe(probe);
-    std::optional<pid_t> pid = child_ ? std::make_optional(child_->pid())
-                                      : this->pid();
-
-    if (probe.type == ProbeType::usdt) {
-      auto aps = attach_usdt_probe(probe, program, pid, usdt_file_activation_);
-      for (auto &ap : aps)
-        ret.emplace_back(std::move(ap));
-
-      return ret;
-    } else if (probe.type == ProbeType::uprobe ||
-               probe.type == ProbeType::uretprobe) {
-      ret.emplace_back(std::make_unique<AttachedProbe>(
-          probe, program, pid, *this, safe_mode_));
-      return ret;
-    } else if (probe.type == ProbeType::watchpoint ||
-               probe.type == ProbeType::asyncwatchpoint) {
-      ret.emplace_back(
-          std::make_unique<AttachedProbe>(probe, program, pid, *this));
-      return ret;
-    } else {
-      ret.emplace_back(
-          std::make_unique<AttachedProbe>(probe, program, pid, *this));
-      return ret;
-    }
-  } catch (const util::EnospcException &e) {
-    // Caller will handle
-    throw e;
-  } catch (const std::exception &e) {
-    LOG(ERROR) << e.what();
-    ret.clear();
+  const auto &program = bytecode.getProgramForProbe(probe);
+  std::optional<pid_t> pid = child_ ? std::make_optional(child_->pid())
+                                    : this->pid();
+
+  if (probe.type == ProbeType::usdt) {
+    auto aps = attach_usdt_probe(probe, program, pid, usdt_file_activation_);
+    for (auto &ap : aps)
+      ret.emplace_back(std::move(ap));
+
+  } else if (probe.type == ProbeType::uprobe ||
+             probe.type == ProbeType::uretprobe) {
+    ret.emplace_back(std::make_unique<AttachedProbe>(
+        probe, program, pid, *this, safe_mode_));
+  } else {
+    ret.emplace_back(
+        std::make_unique<AttachedProbe>(probe, program, pid, *this));
```

Comment:
Continuing from above, `attach_probe()` does not actually attach the probe anymore. Callers have to iterate through the returned vector and call `->attach()`. So probably better to go with static factory method and keep current code structure.

Reply:
See comment above.